src/types/packet.rs

```diff
@@ -142,17 +150,56 @@ pub enum KeyVersion {
     V3 = 3,
     V4 = 4,
     V5 = 5,
+    V6 = 6,
 
     #[num_enum(catch_all)]
     Other(u8),
 }
 
+impl KeyVersion {
+    /// Size of OpenPGP fingerprint in bytes
+    /// (returns 0 for unknown versions)
+    pub const fn fingerprint_len(&self) -> usize {
+        match self {
+            KeyVersion::V2 | KeyVersion::V3 => 16, // MD5
+            KeyVersion::V4 => 20,                  // SHA1
+            KeyVersion::V5 | KeyVersion::V6 => 32, // SHA256
+            KeyVersion::Other(_) => 0,             // FIXME?
```

Comment:
return Option/None ?

Reply:
i'm not sure if there's a more reasonable way to do this than e3f6222e52824a3d1a912246d72a754189f1659d (i have mixed feelings about `expect`ing vs erroring, there)

Reply:
yeah that looks good to me, totally fine to `expect` if you know that the versions are already checked at that point

Reply:
ftr: i replaced the expects (see 53214a9d9e54757b6372c37b8697000cb469e753) based on our brief discussion of the brittleness of this approach